pkg/cmd/roachtest/test_runner.go

```diff
@@ -1105,6 +1105,10 @@ func (r *testRunner) runTest(
 	// deferred function below that a Side-Eye snapshot was taken for a timed out
 	// test.
 	sideEyeTimeoutSnapshotURL := ""
+	// preemptedVMCh listens for preempted VMs found by monitorForPreemptedVMs
+	// which actively polls for VM preemptions.
+	preemptedVMChBuf := max(c.spec.NodeCount, 1) // make sure buffer is at least 1 for unit tests
+	preemptedVMCh := make(chan vm.PreemptedVM, preemptedVMChBuf)
```

Comment:
Another idea I had was that we don't need to communicate preempted VMs at all. Over the course of a long running test, we will hit the API dozens of times, one more isn't a big deal. Open to other alternatives as well.

Reply:
> Over the course of a long running test, we will hit the API dozens of times, one more isn't a big deal. Happy with either solution if we feel adding channels is unnecessary complexity but open to opinions.

It does seem like an unnecessary optimization. Further, you still want to call the API upon failure since your last preemption check could have been several minute ago; what if we had preemptions since then?

Reply:
> what if we had preemptions since then?

Hmm, I don't think that should matter since just one preemption is enough to mark the entire test as a flake. Unless we are keeping track of number of preempted VMs and not just number of tests with preemptions, but as far as I can tell we aren't.

That being said, I agree that it's an unnecessary optimization, I'm down to get rid of it :)

Reply:
> since just one preemption is enough to mark the entire test as a flake

Right. What I meant is, let's say last preemption poll was 4 minutes ago, and a VM of a test, which just failed was preempted in the last minute. Since we omit the preemption check during the post-processing of a test failure, and the next poll won't happen for another minute, we would miss the preemption.

Reply:
If no VMs are found by the polling, it does one more check in post processing:

```
	if len(preemptedVMs) == 0 {
		var err error
		preemptedVMs, err = c.GetPreemptedVMs(ctx, l)
		if err != nil {
			l.Printf("failed to check for preempted VMs:\n%+v", err)
		}
```

But the unclearness around this is probably more reason to get rid of it.

Reply:
> If no VMs are found by the polling, it does one more check in post processing

Right, 🤦